Add v1.3 release notes. (#344)
opea-project/docs: release_notes/v1.3.md

- [Validated Hardware](#validated-hardware)
- [Validated Software](#validated-software)
- [Known Issues](#known-issues)
- [Full Changelogs](#full-changelogs)
- [Contributors](#contributors)

## What's New in OPEA v1.3

This release introduces exciting capabilities, optimizations, and user-centric enhancements:

### Advanced Agent Capabilities
- <b>Multi-Turn Conversation</b>: Enhanced the OPEA agent framework for dynamic, context-aware dialogues. ([GenAIComps#1248](https://github.com/opea-project/GenAIComps/pull/1248)) 
- <b>Finance Agent Example</b>: A financial agent [example](https://github.com/opea-project/GenAIExamples/tree/v1.3/FinanceAgent) for automating financial data aggregation and leveraging LLMs to generate insights, forecasts, and strategic recommendations. ([GenAIExamples#1539](https://github.com/opea-project/GenAIExamples/pull/1539))

### Performance and Scalability
- <b>vLLM Enhancement</b>: Integrated [vLLM](https://github.com/vllm-project/vllm) as the default LLM serving backend for key GenAI examples across Intel® Xeon® processors, Intel® Gaudi® accelerators, and AMD® GPUs. ([GenAIExamples#1436](https://github.com/opea-project/GenAIExamples/pull/1436))

> - <b>Horizontal Pod Scaling(HPA) for IaaS</b>: Added auto-scaling support for cloud-native deployments. ([GenAIInfra#822](https://github.com/opea-project/GenAIInfra/pull/822), [GenAIInfra#770](https://github.com/opea-project/GenAIInfra/pull/770))

Author: Thanks, updated.

- <b>KubeAI Operator for OPEA (Alpha release)</b>: Simplified OPEA inference operations in cloud environment and enabled optimal out-of-the-box performance for specific models and hardware using profiles. ([GenAIInfra#945](https://github.com/opea-project/GenAIInfra/pull/945)) 

### Ecosystem Integrations
- <b>Haystack Integration</b>: Enabled OPEA as a backend of [Haystack](https://haystack.deepset.ai). ([Haystack-OPEA#1](https://github.com/opea-project/Haystack-OPEA#1))
- <b>Cloud Readiness</b>: Expanded automated Terraform deployment for ChatQnA to include support for Azure, and enabled CodeGen deployments on AWS and GCP. ([GenAIExamples#1731](https://github.com/opea-project/GenAIExamples/pull/1731))

### New GenAI Capabilities

> - <b>OPEA Store</b>: Delivered a unified data store access API and a robust data store integration layer that streamlines data store integration. ArangoDB was ported supporting data preparation and retrieval. ([GenAIComps#1528](https://github.com/opea-project/GenAIComps/issues/1528))

Reviewer: I would split this into 2 separate. 1) OPEA Store: Delivered a unified data store Client API for basic CRUD functionality around databases. ArangoDB has been integrated. 
2) I would move ArandoDB [dataprep] and [retrieval] into separate lines under other notable changes/functionality (these are new additions)
These two are independent contributions

Author: Updated, thanks.

### Enhanced Evaluation
- <b>Enhanced Long-Context Model Evaluation</b>: Supported evaluating long-context model on Intel® Gaudi® with vLLM. ([HELMET#20](https://github.com/princeton-nlp/HELMET/pull/20))
- <b>TAG-Bench for SQL Agents</b>: Integrated [TAG-Bench](https://github.com/TAG-Research/TAG-Bench) to evaluate complex SQL query generation ([GenAIEval#230](https://github.com/opea-project/GenAIEval/pull/230)).
- <b>DocSum Support</b>: GenAIEval now supports evaluating the performance of DocSum. ([GenAIEval#252](https://github.com/opea-project/GenAIEval/pull/252))
- <b>Toxicity Detection Evaluation</b>: Introduced a workflow to evaluate the capability of detecting toxic language based on LLMs. ([GenAIEval#241](https://github.com/opea-project/GenAIEval/pull/241))
- <b>Model Card</b>: Added a model card generator for generating reports containing model performance and fairness metrics. ([GenAIEval#236](https://github.com/opea-project/GenAIEval/pull/236))

### Observability
- <b>OpenTelemetry Tracing</b>: Leveraged OpenTelemetry to enable tracing for ChatQnA and AgentQnA along with TGI and TEI. ([GenAIExamples#1542](https://github.com/opea-project/GenAIExamples/issues/1542))

> 

Author: Thanks, updated.

### Better User Experience
- <b>GenAIStudio</b>: Supported drag-and-drop creation of agentic applications. ([GenAIStudio#50](https://github.com/opea-project/GenAIStudio/pull/50))
- <b>Documentation Refinement</b>: Refined READMEs for key examples to help readers easily locate documentation tailored to deployment, customization, and hardware. ([GenAIExamples#1741](https://github.com/opea-project/GenAIExamples/issues/1741))

> - <b>Slimmed Docker Images</b>: Optimized Docker image sizes for faster deployment of GenAI examples. ([GenAIExamples#1585](https://github.com/opea-project/GenAIExamples/pull/1585))

Author: Thanks, updated.

    - Updated README.md of AIPC quick start ([#1578](https://github.com/opea-project/GenAIExamples/pull/1578)) (852bc70)
    - Added short descriptions to the images OPEA publishes on Docker Hub ([#1637](https://github.com/opea-project/GenAIExamples/pull/1637)) (68747a9)

- CI/CD/UT
    - Added UT for rerank finetuning on Gaudi ([#1472](https://github.com/opea-project/GenAIExamples/pull/1472)) (5f4b182)
    - Enabled Gaudi 3, Rocm and Arc on manually release test. ([#1615](https://github.com/opea-project/GenAIExamples/pull/1615)) (63b789a)
    - Enabled base image build in CI/CD ([#1669](https://github.com/opea-project/GenAIExamples/pull/1669)) (2204fe8)
    - ChatQnA run CI with latest base image, group logs in GHA outputs.  ([#1736](https://github.com/opea-project/GenAIExamples/pull/1736)) (c48cd65)
<!-- - Fix Multimodal & ProductivitySuite Issue (#1820) (b80449b) -->

</details>

<details><summary>GenAIComps</summary> 

- Functionalities
    - [agent] Enabled custom prompt for react_llama and react_langgraph ([#1391](https://github.com/opea-project/GenAIComps/pull/1391)) (558a2f6)

>     - [dataprep] Added Multimodal support for Milvus for dataprep component ([#1380](https://github.com/opea-project/GenAIComps/pull/1380)) (006bd91)

Author: Updated, thanks.

<!-- - chatqna: add VLLM_CPU_KVCACHE_SPACE setting (#981) (591772a) -->
</details>

<details><summary>GenAIStudio</summary> 

- Updated studio fe table UI and updated studio be according to the dataprep refactor ([#32](https://github.com/opea-project/GenAIStudio/pull/32)) (1168507)
- [Feat] Added GenAI Studio UI improvement ([#48](https://github.com/opea-project/GenAIStudio/pull/48)) (ad64f7c)
- Enabled LLM Traces for sandbox ([#51](https://github.com/opea-project/GenAIStudio/pull/51)) (df6b73e)
- Migrated to internal k8 mysql and enable deployment package generation for agentqna ([#52](https://github.com/opea-project/GenAIStudio/pull/52)) (0cddbe0)

</details>

## Deprecations
### Deprecated Examples
The following GenAI examples are deprecated, and were removed since OPEA v1.3:
| Example | Migration Solution | Reasons for Deprecation |
|--|--|--|
|[FaqGen](https://github.com/opea-project/GenAIExamples/tree/v1.2/FaqGen)|Use the example [ChatQnA](https://github.com/opea-project/GenAIExamples/tree/v1.3/ChatQnA) instead.| Provide users with a unified chatbot experience and reduce redundancy. |

### Deprecated Docker Images

> The following Docker images are deprecated and were removed since OPEA v1.3:

Reviewer 2: Deprecated images are **not** removed, there just won't be a `1.3` tagged images for them.  Please rephrase e.g. like this:
```suggestion
The following Docker images are deprecated, and not be updated / tagged for OPEA v1.3 release:
```
Please update v1.4 text similarly.

Author: Thanks, updated.

# OPEA Release Notes v1.3
We are excited to announce the release of OPEA version 1.3, which includes significant contributions from the open-source community. This release addresses over 520 pull requests.

> More information about how to get started with OPEA v1.3 can be found on the [Getting Started](https://opea-project.github.io/latest/index.html) page. All project source code is maintained in the [repository](https://github.com/opea-project). To pull Docker images, please access the [Docker Hub](https://hub.docker.com/u/opea). For instructions on deploying Helm Charts, please refer to the [guide](https://github.com/opea-project/GenAIInfra/tree/v1.3/helm-charts#readme).

Author: Thanks. Updated.

## Table of Contents
- [What's New in OPEA v1.3](#whats-new-in-opea-v13)
- [Deprecations](#deprecations)

> - [Dependencies Updates](#dependencies-updates)

Reviewer 3: `[Dependencies Updates](#dependencies-updates)` -> `[Updated Dependencies](#updated-dependencies)`

- [Changes to Default Behavior](#changes-to-default-behavior)
- [Validated Hardware](#validated-hardware)
- [Validated Software](#validated-software)
- [Known Issues](#known-issues)
- [Full Changelogs](#full-changelogs)
- [Contributors](#contributors)

## What's New in OPEA v1.3

This release introduces exciting capabilities, optimizations, and user-centric enhancements:

### Advanced Agent Capabilities
- <b>Multi-Turn Conversation</b>: Enhanced the OPEA agent framework for dynamic, context-aware dialogues. ([GenAIComps#1248](https://github.com/opea-project/GenAIComps/pull/1248)) 

> - <b>Finance Agent Example</b>: A financial agent [example](https://github.com/opea-project/GenAIExamples/FinanceAgent) for automating financial data aggregation and leveraging LLMs to generate insights, forecasts, and strategic recommendations. ([GenAIExamples#1539](https://github.com/opea-project/GenAIExamples/pull/1539))

Reviewer 3: Actually I'd just remove the hyperlink `(https://github.com/opea-project/GenAIExamples/FinanceAgent)` which is broken too, since you already have the link for PR #1539

Author: Thanks. Updated.

> - <b>Finance Agent Example</b>: A financial agent [example](https://github.com/opea-project/GenAIExamples/FinanceAgent) for automating financial data aggregation and leveraging LLMs to generate insights, forecasts, and strategic recommendations. ([GenAIExamples#1539](https://github.com/opea-project/GenAIExamples/pull/1539))

Author: Thanks, updated.

- <b>Enhanced Multimodality</b>: Added support for additional audio file types (.mp3) and supported spoken audio captions with image ingestion. ([GenAIExamples#1549](https://github.com/opea-project/GenAIExamples/issues/1549))
- <b>Struct to Graph</b>: Supported transforming structured data to graphs using Neo4j graph database. ([GenAIComps#1502](https://github.com/opea-project/GenAIComps/pull/1502))
- <b>Text to Graph</b>: Supported creating graphs from text by extracting graph triplets. ([GenAIComps#1357](https://github.com/opea-project/GenAIComps/pull/1357), [GenAIComps#1472](https://github.com/opea-project/GenAIComps/pull/1472))
- <b>Text to Cypher</b>: Supported generating and executing Cypher queries from natural language for graph database retrieval. ([GenAIComps#1319](https://github.com/opea-project/GenAIComps/pull/1319))

### Enhanced Evaluation
- <b>Enhanced Long-Context Model Evaluation</b>: Supported evaluating long-context model on Intel® Gaudi® with vLLM. ([HELMET#20](https://github.com/princeton-nlp/HELMET/pull/20))

> - <b>TAG-Bench for SQL Agents</b>: Integrated [TAG-Bench](https://github.com/TAG-Research/TAG-Bench) to evaluate complex SQL query generation ([GenAIEval#230](https://github.com/opea-project/GenAIEval#230)).

Reviewer 3: Provide the link to the actual PR:
`(https://github.com/opea-project/GenAIEval#230)` -> 
`(https://github.com/opea-project/GenAIEval/pull/230)`

Author: Thanks, updated.

- <b>DocSum Support</b>: GenAIEval now supports evaluating the performance of DocSum. ([GenAIEval#252](https://github.com/opea-project/GenAIEval/pull/252))
- <b>Toxicity Detection Evaluation</b>: Introduced a workflow to evaluate the capability of detecting toxic language based on LLMs. ([GenAIEval#241](https://github.com/opea-project/GenAIEval/pull/241))
- <b>Model Card</b>: Added a model card generator for generating reports containing model performance and fairness metrics. ([GenAIEval#236](https://github.com/opea-project/GenAIEval/pull/236))

### Observability

> - <b>OpenTelemetry Tracing</b>: Leveraged OpenTelemetry to enable tracing for ChatQnA and AgentQnA on TGI and TEI. ([GenAIExamples#1542](https://github.com/opea-project/GenAIExamples/issues/1542))

Reviewer 4: We have OPEA specific tracing. TGI and TEI tracing are from huggingface.  They are no dependency among them, so better to use along with instead of on. 
```suggestion
- <b>OpenTelemetry Tracing</b>: Leveraged OpenTelemetry to enable tracing for ChatQnA and AgentQnA along with TGI and TEI. ([GenAIExamples#1542](https://github.com/opea-project/GenAIExamples/issues/1542))
```

| deepseek-ai/DeepSeek-R1-Distill-Qwen-32B    | ✓         | -        | ✓         | -        | ✓              | -               |
| deepseek-ai/Deepseek-v3                     | ✓         | -        | ✓         | -        | ✓              | -               | 
| Hermes-3-Llama-3.1-8B                       | -         | -        | -          | -        | -              | ✓               |
| ibm-granite/granite-3.2-8b-instruct         | -         | -        | ✓          | -        |                | -               |
| Phi-4-mini                                  | x         | x        | x          | x        | ✓              | -               |
| Phi-4-multimodal-instruct                   | x         | x        | x          | x        | ✓              | -               |
| mistralai/Mistral-Small-24B-Instruct-2501   | ✓         | -        | ✓         | -        | ✓              | -               |
| mistralai/Mistral-Large-Instruct-2411       | x         | -        | ✓         | -        | ✓              | -               |

(✓: supported; -: not validated; x: unsupported)

> [All Supported LLM Models](https://github.com/opea-project/GenAIComps/tree/v1.3/comps/llms/src/text-generation#validated-llm-models)

Reviewer 3: besides this one: https://github.com/opea-project/GenAIComps/tree/main/comps/llms/src which doesn't have a table for supported LLMs, I'm not sure if we have such a document.
Should we just remove this link altogether?

> [All Supported LLM Models](https://github.com/opea-project/GenAIComps/tree/v1.3/comps/llms/src/text-generation#validated-llm-models)

Reviewer 5: @mkbhanda Can we expand this model support matrix with a ROCm/Instinct column? AMD team can provide the information.

> [All Supported LLM Models](https://github.com/opea-project/GenAIComps/tree/v1.3/comps/llms/src/text-generation#validated-llm-models)

Author: I did find a table for all supported LLMs in READMEs. I agree with @ashahba to remove the link of all supported LLMs.

### Newly Supported Hardware
- AMD® GPU using AMD® ROCm™ for 9 examples. ([GenAIExamples#1613](https://github.com/opea-project/GenAIExamples/pull/1613) and 8 more.)

### Other Notable Changes

Expand the following lists to read:

<details><summary>GenAIExamples</summary> 

- Functionalities
    - [AgentQnA] Added web search tool support and simplify the run instructions. (#1656) (e8f2313)
    - [ChatQnA] Added support for latest deepseek models on Gaudi (#1491) (9adf7a6)
    - [EdgeCraftRAG] A sleek new UI based on Vue and Ant Design for enhanced user experience, supporting concurrent multi-requests on vLLM, JSON pipeline configuration, and API-based prompt modification. (#1665) (5a50ae0)
    - [EdgeCraftRAG] Supported multi-card deployment of Intel ARC GPU for vllm inference (#1729) (1a0c5f0)
    - [FaqGen] Merged FaqGen into ChatQnA for unified Chatbot experience. (#1654) (6d24c1c)

- Benchmark
    - [ChatQnA] Provided unified scalable deployment and benchmarking support for examples (#1315) (ed16308)

- Deployment
    - Sync values yaml file for 1.3 release (#1748) (46ebb78)

> - Bug Fixes

Author: Thanks. I will use static links instead.

|[opea/chathistory-mongo-server](https://hub.docker.com/r/opea/chathistory-mongo-server)|Use [opea/chathistory-mongo](https://hub.docker.com/r/opea/chathistory-mongo) instead.| Follow the OPEA naming rules |
|[opea/faqgen](https://hub.docker.com/r/opea/faqgen)|Use [opea/chatqna](https://hub.docker.com/r/opea/chatqna) or [opea/chatqna-without-rerank](https://hub.docker.com/r/opea/chatqna-without-rerank) instead.| FaqGen is deprecated. |
|[opea/faqgen-ui](https://hub.docker.com/r/opea/faqgen-ui)|Use [opea/chatqna-ui](https://hub.docker.com/r/opea/chatqna-ui) instead.| FaqGen is deprecated. | 
|[opea/faqgen-react-ui](https://hub.docker.com/r/opea/faqgen-react-ui)|Use [opea/chatqna-ui](https://hub.docker.com/r/opea/chatqna-ui) instead.| FaqGen is deprecated. |
|[opea/feedbackmanagement](https://hub.docker.com/r/opea/feedbackmanagement)|Use [opea/feedbackmanagement-mongo](https://hub.docker.com/r/opea/feedbackmanagement-mongo) instead.| Follow the OPEA naming rules |
|[opea/promptregistry-mongo-server](https://hub.docker.com/r/opea/promptregistry-mongo-server)|Use [opea/promptregistry-mongo](https://hub.docker.com/r/opea/promptregistry-mongo) instead.| Follow the OPEA naming rules |

> The following Docker images are deprecated and will be removed since OPEA v1.4:

Author: Thanks. Updated.

|[DocIndexRetriever](https://github.com/opea-project/GenAIExamples/tree/v1.3/DocIndexRetriever)|environment variable|`LLM_SERVICE_HOST_IP`|Removed due to no uses.|
|[GraphRAG](https://github.com/opea-project/GenAIExamples/tree/v1.3/GraphRAG)|environment variable|`MAX_OUTPUT_TOKENS`|Instead, it has been split into two new environment variables: `MAX_INPUT_TOKENS` (default: 4096) and `MAX_TOTAL_TOKENS` (default: 8192) to control the maximum token limits.|

### Deprecated GenAIComps Parameters
| Component | Parameter | Migration Solution |
|--|--|--|
|[agent](https://github.com/opea-project/GenAIComps/tree/v1.3/comps/agent)|`with_store` of `agent_config` in the Assistants APIs|Its functionality is now fully covered by the new `memory_type` parameter. In v1.3, please use `"with_memory": true` and `"memory_type": persistent` as its replacement. The `with_memory` parameter in `agent_config` of APIs is now enabled by default (true) for enabling multi-turn conversations. Please refer to the [guide](https://github.com/opea-project/GenAIComps/blob/v1.3/comps/agent/src/README.md#15-agent-memory) for more details.|

> ## Dependencies Updates

Reviewer 3: `Dependencies Updates` -> `Updated Dependencies`

- <b>CodeGen using RAG and Agent</b>: Leveraged RAG and code agent to provide an additional layer of intelligence and adaptability for CodeGen example. ([GenAIExamples#1757](https://github.com/opea-project/GenAIExamples/pull/1757))
- <b>Enhanced Multimodality</b>: Added support for additional audio file types (.mp3) and supported spoken audio captions with image ingestion. ([GenAIExamples#1549](https://github.com/opea-project/GenAIExamples/issues/1549))
- <b>Struct to Graph</b>: Supported transforming structured data to graphs using Neo4j graph database. ([GenAIComps#1502](https://github.com/opea-project/GenAIComps/pull/1502))
- <b>Text to Graph</b>: Supported creating graphs from text by extracting graph triplets. ([GenAIComps#1357](https://github.com/opea-project/GenAIComps/pull/1357), [GenAIComps#1472](https://github.com/opea-project/GenAIComps/pull/1472))
- <b>Text to Cypher</b>: Supported generating and executing Cypher queries from natural language for graph database retrieval. ([GenAIComps#1319](https://github.com/opea-project/GenAIComps/pull/1319))

### Enhanced Evaluation

> - <b>Enhanced Long-Context Model Evaluation</b>: Supported evaluating long-context model on Intel® Gaudi® with vLLM. ([HELMET#20](https://github.com/princeton-nlp/HELMET/pull/20))

Author: It began as an OPEA goal, and ultimately landed within HELMET to meet the requirement.
